src/components/NoteCards/NoteCards.jsx

```diff
@@ -1,44 +1,129 @@
 import { Palette, Label } from "../index";
-import { useModal } from "../../context/index";
+import { useModal, useNotes } from "../../context/index";
 import { ModalInput } from "../ModalInput/ModalInput";
+import { useEffect } from "react";
+import { addNote, updateNote } from "../../services/noteServices";
 
 export const NoteCards = ({ data }) => {
   const { modalState, modalDispatch } = useModal();
+  const { notes, noteData, noteDispatch, currNote } = useNotes();
+  // console.log(notes);
+  // console.log(noteData);
+  const { isPinned, isTrashed, isArchived, noteColor } = noteData;
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await updateNote(noteData._id, noteData);
+        noteDispatch({ type: "NOTE_UPDATED", payload: data.notes });
+        // console.log(data.notes);
+      } catch (error) {
+        // console.log(error);
+      }
+    })();
+  }, [noteData]);
 
   return (
     <>
       <div className="notes-keeping-area ">
-        {data &&
-          data.map((dataItem) => (
-            <div className="note-input-container grid-item-note">
+        {notes.length !== 0 &&
+          notes.map((dataItem) => (
+            <div
+              className="note-input-container grid-item-note "
+              style={{ backgroundColor: dataItem.noteColor }}
```

Comment:
Refrain from using inline styles. Use css classes instead

Reply:
It's used to change the container color, when user choose color from palette.